src/apps/packet_filter/README.md.src

```diff
@@ -54,15 +74,11 @@ is allowed.
 
 — Key **state_track**
 
-*Optional*. Tracks connections state in a named connection table.  If used
-inside a rule, every packet that passes this specific rule is tracked.
-If used 'outside' any rule, packets passing _any_ rule on this app will be
-tracked.
+*Optional*. A string naming a connection table. If set, packets passing
+the rule are tracked in the specified connection table.
 
 — Key **state_check**
 
-*Optional*. Checks if a packet belongs to a connection in a named
-connection table.  If used inside a rule, a packet must belong to an
-existing connection in addition to any other condition in the rule to pass.
-If used 'outside' any rule, if a packet belongs to an existing connection
-it's enough to pass the app.
+*Optional*. A string denoting a connection table. If set, packets must
```

Comment:
More than once I've found the iptables connection tracking behaviour to be limiting.  In short, it tracks _all_ packets on a single table and then you can write rules that check a packet.  In this implementation, both the tracking and checking can be used on a rule or globally, and also to use separate, named tables.  This allows the user to only track packets that pass a certain rule, or to discriminate different connections in different tracking tables.

If we wanted to only emulate iptables, we could use only the 'inside rule' `state_check`, and an implicit global `state_track`, both storing in a single table.